File: book/petrol-pump/chapter2/section7/Nozzle8888.java
package chapter2.section7;

import pump.*;
import sodium.*;

public class Nozzle8888 implements Pump
{
    public Outputs create(Inputs inputs)
    {
        Behavior<UpDown> nozzle1 = inputs.eNozzle1.hold(UpDown.DOWN);
        Behavior<UpDown> nozzle2 = inputs.eNozzle2.hold(UpDown.DOWN);
        Behavior<UpDown> nozzle3 = inputs.eNozzle3.hold(UpDown.DOWN);
        return new Outputs()
            .setPriceLCD1(nozzle1.map(u ->
                u.equals(UpDown.UP) ?  "8.8.8.8." : ""))
            .setPriceLCD2(nozzle2.map(u ->
                u.equals(UpDown.UP) ?  "8.8.8.8." : ""))
            .setPriceLCD3(nozzle3.map(u ->
                u.equals(UpDown.UP) ?  "8.8.8.8." : ""));
    }
}

File: c#/Book/PetrolPump/PetrolPump/PetrolPump/Chapter2/Section8/Nozzle8888.cs
using PetrolPump.Pump;

using Sodium;

namespace PetrolPump.Chapter2.Section8
{

public class Nozzle8888 : IPump
{
    public Outputs Create(Inputs inputs) {
        Behavior<UpDown> nozzle1 = inputs.ENozzle1.Hold(UpDown.DOWN);
        Behavior<UpDown> nozzle2 = inputs.ENozzle2.Hold(UpDown.DOWN);
        Behavior<UpDown> nozzle3 = inputs.ENozzle3.Hold(UpDown.DOWN);
        return new Outputs()
            .SetPriceLcd1(nozzle1.Map(u =>
                u.Equals(UpDown.UP) ?  "8.8.8.8." : ""))
            .SetPriceLcd2(nozzle2.Map(u =>
                u.Equals(UpDown.UP) ?  "8.8.8.8." : ""))
            .SetPriceLcd3(nozzle3.Map(u =>
                u.Equals(UpDown.UP) ?  "8.8.8.8." : ""));
    }
}
}
